src/plugins/foxPage/components/AssetActions.tsx

```diff
@@ -9,39 +11,78 @@ import {
   Tabs,
   Text as CText,
 } from '@chakra-ui/react'
+import { AssetId } from '@shapeshiftoss/caip'
+import { foxyAddresses } from '@shapeshiftoss/investor-foxy'
+import { FoxyPath } from 'features/defi/providers/foxy/components/FoxyManager/FoxyManager'
+import qs from 'qs'
 import { useTranslate } from 'react-polyglot'
+import { useHistory, useLocation } from 'react-router'
 import { AssetIcon } from 'components/AssetIcon'
 import { Card } from 'components/Card/Card'
 import { Text } from 'components/Text/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { useModal } from 'hooks/useModal/useModal'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { useGetAssetDescriptionQuery } from 'state/slices/assetsSlice/assetsSlice'
+import { trimWithEndEllipsis } from 'state/slices/portfolioSlice/utils'
+import { selectAssetById } from 'state/slices/selectors'
+import { selectAccountIdsByAssetId } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import { FoxAssetId, TrimmedDescriptionLength } from '../constants'
 
 type FoxTabProps = {
-  assetIcon: string
-  assetSymbol: string
-  description: string
-  buyCTA: string
-  sellCTA: string
-  onReceiveClick: () => void
-  onBuyClick: () => void
+  assetId: AssetId
 }
 
-export const AssetActions = ({
-  assetIcon,
-  assetSymbol,
-  description,
-  buyCTA,
-  sellCTA,
-  onReceiveClick,
-  onBuyClick,
-}: FoxTabProps) => {
+const GetFoxyModalRoute = FoxyPath.Overview
+const GetFoxCoinbaseExternalUrl = `https://www.coinbase.com/price/fox-token`
+
+export const AssetActions: React.FC<FoxTabProps> = ({ assetId }) => {
   const translate = useTranslate()
+  const history = useHistory()
+  const location = useLocation()
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const { description } = asset || {}
+  const query = useGetAssetDescriptionQuery(assetId)
+  const isLoaded = !query.isLoading
+  const trimmedDescription = trimWithEndEllipsis(description, TrimmedDescriptionLength)
+  const isFoxAsset = assetId === FoxAssetId
+
+  const accountIds = useAppSelector(state => selectAccountIdsByAssetId(state, { assetId }))
+  const accountId = accountIds?.[0]
+  const {
+    state: { isConnected },
+    dispatch,
+  } = useWallet()
+  const { receive } = useModal()
+  const handleWalletModalOpen = () =>
+    dispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
+  const handleReceiveClick = () =>
+    isConnected ? receive.open({ asset, accountId }) : handleWalletModalOpen()
+
+  const onGetAssetClick = () => {
+    history.push({
+      pathname: GetFoxyModalRoute,
+      search: qs.stringify({
+        chain: asset.chain,
```

Comment:
```suggestion
        chain: asset.chain,
```

chain is deprecated - please don't add it in new code. what do we have to do to use `chainId` here?

Reply:
Asset's have a chainId property now, we can just do:

```suggestion
        chainId: asset.chainId,
```

and update wherever it is consumed.

Reply:
We'll need to update it everywhere in DeFi beforehand, otherwise this will be a ~25 files diff just for the the change of queryparam from chain to chainId.
I'll PR it and after that's merged, you'll just be able to commit this suggestion @chxash

Reply:
Note that in addition to the refactor in DeFi, usages will also have to be removed in all current `history.push()` calls, that all include a `chain` param

Reply:
@chxash https://github.com/shapeshift/web/pull/1903 just got in, you can now rebase this and use `chainId` instead

Reply:
Done ! thanks for the PR @gomesalexandre